subprojects/base-services/src/main/java/org/gradle/internal/service/DefaultServiceRegistry.java

```diff
@@ -76,10 +74,30 @@
  */
 public class DefaultServiceRegistry implements ServiceRegistry, Closeable {
     private final static ServiceRegistry[] NO_PARENTS = new ServiceRegistry[0];
+    private final static ServiceProvider[] NO_DEPENDENTS = new ServiceProvider[0];
+    private final static Object[] NO_PARAMS = new Object[0];
 
     private static final ConcurrentMap<Type, BiFunction<ServiceProvider, LookupContext, Provider>> SERVICE_TYPE_PROVIDER_CACHE = new ConcurrentHashMap<Type, BiFunction<ServiceProvider, LookupContext, Provider>>();
     private final Map<Type, ServiceProvider> providerCache = new IdentityHashMap<Type, ServiceProvider>();
 
+    private final static Object lookup = createLookup();
+
+
+    /**
+     * This method is intentionally weakly typed, so that we don't depend on JDK 7+ types.
+     */
+    static Object getLookup() {
```

Comment:
I removed it. Basically `MethodHandle` won't let you call `protected` or `private` methods, if the lookup object that was used is not "in scope" of the target object. Since this method always returns the lookup object for `DefaultServiceRegistry`, it is not allowed to access protected methods of subclasses of the registry in any case.